### ISSUE TITLE
Do not want to use Entity for my classes.

### ISSUE THREAD

**Issue by skalinkin** [RobThree/MongoRepository#25]
Hi Guys, 
I would like to have my application abstracted from data store layer. But in order to use Generic Repository I have to inherit my entities from Entity class. I would like to have any references to your library from my application but inject data layer using container. Do you have any plans to support entities based on DataAnnotation attributes for example, or something else?

**Comment by RobThree** [comment]
No. Best I can do is advise you to use IEntity instead of Entity. We will, [for 2.0](https://github.com/RobThree/MongoRepository/issues/4#issuecomment-165779769) try to [move the interfaces](https://github.com/RobThree/MongoRepository/issues/19) to a separate assembly / package. Other than that we have no concrete plans for moving even more loosly coupled by using attributes *yet*.
